clang/lib/Sema/SemaTemplate.cpp

```diff
@@ -3331,6 +3343,65 @@ checkBuiltinTemplateIdType(Sema &SemaRef, BuiltinTemplateDecl *BTD,
     QualType HasNoTypeMember = Converted[2].getAsType();
     return HasNoTypeMember;
   }
+  case BTK__builtin_dedup_pack: {
+    assert(Converted.size() == 1 && "__builtin_dedup_pack should be given "
+                                    "a parameter pack");
```

Comment:
Should that not be a diagnostic?

Reply:
Since we require the arguments to match first, I don't think we would ever get anything other than a pack here. Other builtins follow the same pattern. Am I wrong here? Could you provide an example where this happens?

Reply:
IIRC, this should be caught by SemaChecking based on the builtin definition.  See the change to `[clang/include/clang/Basic/BuiltinTemplates.td](https://github.com/llvm/llvm-project/pull/106730/files#diff-d3763371ff5a25641558b8e3a2d166846d2dbfc8738da3f37f80b49a397b1f9d)`, so normal template-function-call checking should take care of these.